internal/model/decoder.go

```diff
@@ -17,17 +17,20 @@ package model
 // MetricsDecoder interface to decode bytes into protocol-specific data model.
 type MetricsDecoder interface {
 	// DecodeMetrics decodes bytes into protocol-specific data model.
+	// If error is not nil returned interface cannot be used.
```

Comment:
```suggestion
	// If the returned error is not nil, the returned interface cannot be used.
```

I wonder if implementation shouldn't be required to return nil for the interfaces in this case?